drivers/sx126x/sx126x_radio_hal.c

```diff
@@ -0,0 +1,610 @@
+/*
+ * Copyright (C) 2023
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sx126x
+ * @{
+ * @file
+ * @brief       Radio HAL implementation for the Semtech SX126x
+ *
+ * @author      Klim Evdokimov <klimevdokimov@mail.ru>
+ * @author      Fabian Hüßler <fabian.huessler@ml-pa.com
+ * @}
+ */
+
+#include <errno.h>
+#include <stdio.h>
+
+#include "macros/utils.h"
+#include "modules.h"
+#if IS_USED(MODULE_SX126X_IEEE802154)
+#include "net/ieee802154/radio.h"
+#include "net/ieee802154/submac.h"
+#endif
+#include "od.h"
+
+#include "sx126x.h"
+#include "sx126x_params.h"
+#include "sx126x_internal.h"
+
+#define ENABLE_DEBUG 1
+#include "debug.h"
+
+#ifndef SX126X_HAL_CHAN_BASE
+#  define SX126X_HAL_CHAN_BASE          (868300000LU)
+#endif
+
+#ifndef SX126X_HAL_CHAN_SPACING
+#  define SX126X_HAL_CHAN_SPACING       (200000LU)
+#endif
+
+#ifndef SX126X_CHAN_MIN
+#  define SX126X_CHAN_MIN               (0)
+#endif
+
+#ifndef SX126X_CHAN_MAX
+#  define SX126X_CHAN_MAX               (8)
+#endif
+
+#ifndef SX126X_POWER_MIN
+#  define SX126X_POWER_MIN              (-9)
+#endif
+
+#ifndef SX126X_POWER_MAX
+#  define SX126X_POWER_MAX              (22)
+#endif
+
+#ifndef SX126X_POWER_DEFAULT
+#  define SX126X_POWER_DEFAULT          (0)
+#endif
+
+#ifndef SX126X_CHANNEL_DEFAULT
+#  define SX126X_CHANNEL_DEFAULT        (0)
+#endif
+
+#ifndef SX126X_CHANNEL_PAGE_DEFAULT
+#  define SX126X_CHANNEL_PAGE_DEFAULT   (0)
+#endif
+
+#define SX126X_BUFFER_SIZE              (256)
+#define SX126X_RX_BUFFER_OFFSET         (0)
+#define SX126X_TX_BUFFER_OFFSET         ((SX126X_BUFFER_SIZE) / 2)
+
+#if IS_USED(MODULE_SX126X_IEEE802154)
+
+static int _set_state(sx126x_t *dev, sx126x_state_t state);
+static int _get_state(sx126x_t *dev, void  *val);
+
+static bool _l2filter(ieee802154_dev_t *hal, uint8_t *mhr)
+{
+    sx126x_t *dev = hal->priv;
+    uint8_t dst_addr[IEEE802154_LONG_ADDRESS_LEN];
+    le_uint16_t dst_pan;
+    uint8_t pan_bcast[] = IEEE802154_PANID_BCAST;
+
+    int addr_len = ieee802154_get_dst(mhr, dst_addr, &dst_pan);
+
+    if ((mhr[0] & IEEE802154_FCF_TYPE_MASK) == IEEE802154_FCF_TYPE_BEACON) {
+        if ((memcmp(&dev->pan_id, pan_bcast, 2) == 0)) {
+            DEBUG("[sx126x] hal: beacon\n");
+            return true;
+        }
+    }
+    /* filter PAN ID */
+    /* Will only work on little endian platform (all?) */
+
+    if ((memcmp(pan_bcast, dst_pan.u8, 2) != 0) &&
```

Comment:
Not that I care much about Big Endian MCUs (might as well care about Bigfoot), but you really don't need `memcpy()` here

```suggestion
    if (pan_bcast != byteorder_ltohs(dst_pan)) &&
```